Templates/cmake/README.md

```diff
@@ -0,0 +1,125 @@
+# CMake Template Examples for OneAPI
+
+The CMake projects in this code repository demonstrate use of CMake to build
+simple programs with OneAPI compilers for various common scenarios.
+
+
+| Area                      | Description
+|:---                       |:---
+| What you will learn       | How to use CMake to build projects with OneAPI compilers.
+| Time to complete          | 10 minutes
+| Category                  | Tutorial
+
+## Purpose
+
+The CMake projects in this code repository demonstrate use of CMake to build
+simple programs with OneAPI Compilers for various common scenarios. Examples are divided
+-into directories for C, C++, and Fortran. SYCL examples are included at the
+top level. SYCL examples are implemented entirely in C++.
+
+## Prerequisites
+
+| Optimized for             | Description
+|:---                       |:---
+| OS                        | Windows, Linux
+| Hardware                  | N/A
+| Software                  | CMake, OneAPI Compilers
+
+## Key Implementation Details
+
+
+Each language specific directory are contains few sample examples such as OpenMP, IPO.
+
+The examples in this directory are structured as a collection of independent
+projects, rather as a large single project. This way, any example can be copied
+into a separate subdirectory, compiled, run, and used as the basis for a new project.
+
+The top level CMakeLists.txt includes projects in all the sub-directories.  To
+build all of the examples, create a build directory and generate the project as
+usual.  
+ 
+For example in Linux,
+
+    $ mkdir build
+    $ cd build
+    $ CC=icx CXX=icpx FC=ifx cmake ..

```

Comment:
Yes, We can do that but that requires changing CMakeLists.txt by user and thought command line is easier. Another reason is that user will be forced to think/ensure the availability of oneAPI compilers. I can change it as you suggested if these are not an issue..

Reply:
My thoughts are that: This is a OneAPI compiler cmake tutorial, so I would assume that these compilers are available to the user. Maybe it would be worth to only set the compiler that is required by the current project being compiled (so that cmake does not check if FC is working when the project is compiled with the SYCL compiler.
I find the error message ok when cmake does not find the compiler through: set(CMAKE_CXX_COMPILER icpx)

Reply:
Ok, another thought I just realized is that we support a combination of drivers in oneAPI
for eg:
Just for C++, 
In windows:
     native style: icx
     GNU style: icpx
In Linux:
     native style: icpx
     windows style: icx-cl
We have similar combination with other C/SYCL as well

Do we still want to have it added to CMakeLists.txt instead of commandline?

Reply:
My feeling is that:
1/ We want users to use icpx on Linux and icx-cl on Windows. Let's make it clear by having it directly in the cmake file. The only reason one would want to change the compiler being used is if they want to use specific compiler flags. In this situation, they'll need to modify the cmake anyway, so they can change the compiler if needed.
2/ ipcx on Windows does not really work (through cmake I mean), I suspect icx-cl has similar issues on Linux.
3/ The flags you use in your current cmake would work for all these combinations? 

For all these reasons, I'd say we should set icpx on Linux and icx-cl on Windows (and test both), what are your thoughts?

Reply:
Ok then, I will update the scripts and document